README.md

```diff
@@ -0,0 +1,71 @@
+# Proof Generator
+
+A proof generator for cairo programs.
+
+### Credits
+
+This repository uses [stone-prover](https://github.com/starkware-libs/stone-prover) under the hood to generate proofs and uses [cairo-vm](https://github.com/lambdaclass/cairo-vm) to run, compile. The bootloader is taken from [bootloader](https://github.com/Moonsong-Labs/cairo-bootloader.git) repository.
+
+## Overview
+
+You can generate proofs for cairo programs here. The commands available are documented here.
+There are 2 ways to genrate proofs:
+
+1. First bootloading the program and then proving the execution of the bootloader. This is the recommended way and generates smaller proofs. Check out a simple bootloader [here](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/cairo/bootloaders/simple_bootloader/simple_bootloader.cairo_)
+2. Proving the execution of the program directly.
+
+## Installation
+
+Run the following command to install the dependencies:
+
+```shell
+./install.sh
+cargo build
+```
+
+## Files
+
+`compiled.json`: The compiled Cairo program.
+`input.json`: Input data for the Cairo program.
+`public_input.json`: Public inputs for the proof generation.
+`private_input.json`: Private inputs for the proof generation.
+`trace.bin`: Execution trace of the Cairo program.
+`memory.bin`: Memory dump of the Cairo program execution.
+`proof.json`: The generated STARK proof.
+`pie.zip`: Cairo PIE (Program Independent Executable) that can be generated by running the program and can be used re run the file in standardised way without inputs.
```

Comment:
we should add that this is only for debugging purposes or if you're using SHARP

Reply:
actually, maybe we can remove it as well? how do you suggest to use the PIE for debugging?

Reply:
Sometimes, if you have errrors while running the pie, they could potentially debugged faster than facing that while prooving it with sharp. That happended once with me if I remember correctly.

- Added the note in readme